iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java

```diff
@@ -0,0 +1,105 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import java.util.Collection;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public class ExportersTest extends AbstractConvertersTest {
+
+    private static final String FOO_TST = "foo.tst";
+    private static final String WORK_FOO_TST = "/work/" + FOO_TST;
+
+    private final Exporter testExporter = new TestExporter();
+    private final ExportersLoader loader = new ExportersLoaderList(testExporter);
+
+    @Test
+    public void getFormats() {
+        Collection<String> formats = Exporters.getFormats(loader);
+        assertEquals(1, formats.size());
+        assertTrue(formats.contains(TEST_FORMAT));
+    }
+
+    @Test
+    public void getExporter() {
+        Exporter exporter = Exporters.getExporter(loader, TEST_FORMAT);
+        assertNotNull(exporter);
+        assertEquals(testExporter, exporter);
+    }
+
+    @Test
+    public void getNullExporter() {
+        Exporter exporter = Exporters.getExporter(loader, UNSUPPORTED_FORMAT);
+        assertNull(exporter);
+    }
+
+    @Test
+    public void createDataSource1() throws IOException {
+        Files.createFile(fileSystem.getPath(WORK_FOO_TST));
+        DataSource dataSource = Exporters.createDataSource(fileSystem.getPath("/work/"), "foo", null);
+        assertTrue(dataSource.exists(FOO_TST));
+    }
+
+    @Test
+    public void createDataSource2() throws IOException {
+        Files.createFile(fileSystem.getPath(WORK_FOO_TST));
+        DataSource dataSource = Exporters.createDataSource(path, null);
+        assertTrue(dataSource.exists(FOO_TST));
+    }
+
+    @Test
+    public void createDataSource3() throws IOException {
+        Files.createFile(fileSystem.getPath(WORK_FOO_TST));
+        DataSource dataSource = Exporters.createDataSource(path);
+        assertTrue(dataSource.exists(FOO_TST));
+    }
+
+    @Test
+    public void failExport() {
+        expected.expect(PowsyblException.class);
+        expected.expectMessage("Export format " + UNSUPPORTED_FORMAT + " not supported");
+        Exporters.export(loader, UNSUPPORTED_FORMAT, null, null, Exporters.createDataSource(path));
+    }
+
+    @Test
+    public void export1() throws IOException {
+        DataSource dataSource = Exporters.createDataSource(path);
+        Exporters.export(loader, TEST_FORMAT, null, null, dataSource);
+        assertEquals(Byte.BYTES, dataSource.newInputStream(null, EXTENSION).read());
+    }
+
+    @Test
+    public void export2() throws IOException {
+        Exporters.export(loader, TEST_FORMAT, null, null, path);
+        DataSource dataSource = Exporters.createDataSource(path);
+        assertEquals(Byte.BYTES, dataSource.newInputStream(null, EXTENSION).read());
+    }
+
+    @Test
+    public void export3() throws IOException {
+        Path dir = Files.createTempDirectory("tmp-export");
```

Comment:
Files.createTempDirectory does not use jimfs  but create a temp file on the local filesystem. Is it possible to replace by an in memory one?

Reply:
The tested method needs the String of the directory name, not a path (the path is deduced by `Paths.get` inside the method), so I don't think so...